### PR TITLE
WIP test using old code

### PR DESCRIPTION
**Pull request by lwasser**
This is a PR which i believe has the older code. I was able to "deploy" locally but this deployment did NOT update the envt on the hub so i'm not sure what i did. Where does the docker container get updates in this? i thought i understood but now i'm confused again. i updated the tag on the ea envt version so in theory it would have created a new hub image. 

```
(earth-analytics-python) CIRES-EL-LM-020:hub-ops leahwasser$ python deploy.py --no-setup --build edsc-hub
Do not need to rebuild image, using earthlabhubops/ea-k8s-user-edsc-hub:df3c4bb
(earth-analytics-python) CIRES-EL-LM-020:hub-ops leahwasser$ helm init --client-only
$HELM_HOME has been configured at /Users/leahwasser/.helm.
Not installing Tiller due to 'client-only' flag having been set
(earth-analytics-python) CIRES-EL-LM-020:hub-ops leahwasser$   helm repo add jupyterhub https://jupyterhub.github.io/helm-chart

"jupyterhub" has been added to your repositories
(earth-analytics-python) CIRES-EL-LM-020:hub-ops leahwasser$ 
(earth-analytics-python) CIRES-EL-LM-020:hub-ops leahwasser$ helm repo update
Hang tight while we grab the latest from your chart repositories...
...Skip local chart repository
...Successfully got an update from the "jupyterhub" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete.
(earth-analytics-python) CIRES-EL-LM-020:hub-ops leahwasser$ python deploy.py --build --push --deploy edsc-hub
Activated service account credentials for: [travis-deployer@ea-jupyter.iam.gserviceaccount.com]
Fetching cluster endpoint and auth data.
kubeconfig entry generated for jhub.


Updates are available for some Cloud SDK components.  To install them,
please run:
  $ gcloud components update



To take a quick anonymous survey, run:
  $ gcloud survey

$HELM_HOME has been configured at /Users/leahwasser/.helm.

Tiller (the Helm server-side component) has been updated to gcr.io/kubernetes-helm/tiller:v2.16.7 .
deployment "tiller-deploy" successfully rolled out
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
Do not need to rebuild image, using earthlabhubops/ea-k8s-user-edsc-hub:df3c4bb
Using earthlabhubops/ea-k8s-user-edsc-hub:df3c4bb as user image for edsc-hub
INFO:root:Executing helm dep up
Hang tight while we grab the latest from your chart repositories...
...Unable to get an update from the "local" chart repository (http://127.0.0.1:8879/charts):
	Get "http://127.0.0.1:8879/charts/index.yaml": dial tcp 127.0.0.1:8879: connect: connection refused
...Successfully got an update from the "jupyterhub" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete.
Saving 1 charts
Downloading jupyterhub from repo https://jupyterhub.github.io/helm-chart
Deleting outdated charts
INFO:root:Executing helm upgrade --install --namespace edsc-hub edsc-hub hub-charts/edsc-hub --force --wait --timeout 600 --cleanup-on-fail -f secrets/edsc-hub.yaml --set-string jupyterhub.singleuser.image.tag=df3c4bb
Release "edsc-hub" has been upgraded.
LAST DEPLOYED: Wed Jun 24 19:25:45 2020
NAMESPACE: edsc-hub
STATUS: DEPLOYED

RESOURCES:
==> v1/ClusterRole
NAME                                   AGE
edsc-hub-user-scheduler-complementary  23d

==> v1/ClusterRoleBinding
NAME                                   AGE
edsc-hub-user-scheduler-base           23d
edsc-hub-user-scheduler-complementary  23d

==> v1/ConfigMap
NAME            DATA  AGE
hub-config      4     23d
user-scheduler  1     23d

==> v1/DaemonSet
NAME                     DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
continuous-image-puller  2        2        2      2           2          <none>         23d

==> v1/Deployment
NAME            READY  UP-TO-DATE  AVAILABLE  AGE
hub             1/1    1           1          23d
proxy           1/1    1           1          23d
user-scheduler  2/2    2           2          23d

==> v1/PersistentVolumeClaim
NAME        STATUS  VOLUME                                    CAPACITY  ACCESS MODES  STORAGECLASS  AGE
hub-db-dir  Bound   pvc-ea2fac4e-a43b-11ea-8c26-42010a80013d  1Gi       RWO           standard      23d

==> v1/Pod(related)
NAME                             READY  STATUS   RESTARTS  AGE
continuous-image-puller-4j2zf    1/1    Running  0         23d
continuous-image-puller-xqdb7    1/1    Running  0         23d
user-scheduler-6d7ccf8d6d-6t9l2  1/1    Running  0         23d
user-scheduler-6d7ccf8d6d-c49ml  1/1    Running  0         23d
hub-684787bf56-8s84b             1/1    Running  0         8d
proxy-58bbc859b8-z54kw           1/1    Running  0         23d

==> v1/Role
NAME  AGE
hub   23d

==> v1/RoleBinding
NAME  AGE
hub   23d

==> v1/Secret
NAME        TYPE    DATA  AGE
hub-secret  Opaque  2     23d

==> v1/Service
NAME          TYPE       CLUSTER-IP     EXTERNAL-IP  PORT(S)         AGE
hub           ClusterIP  10.31.247.198  <none>       8081/TCP        23d
proxy-api     ClusterIP  10.31.240.229  <none>       8001/TCP        23d
proxy-public  ClusterIP  10.31.247.4    <none>       443/TCP,80/TCP  23d

==> v1/ServiceAccount
NAME            SECRETS  AGE
hub             1        23d
user-scheduler  1        23d

==> v1/StatefulSet
NAME              READY  AGE
user-placeholder  0/0    23d

==> v1beta1/Ingress
NAME        HOSTS                     ADDRESS  PORTS  AGE
jupyterhub  hub.earthdatascience.org  80, 443  23d

==> v1beta1/PodDisruptionBudget
NAME              MIN AVAILABLE  MAX UNAVAILABLE  ALLOWED DISRUPTIONS  AGE
hub               1              N/A              0                    23d
proxy             1              N/A              0                    23d
user-placeholder  0              N/A              0                    23d
user-scheduler    1              N/A              1                    23d


INFO:root:Waiting for all deployments to be up and running
INFO:root:Executing and capturing kubectl --namespace edsc-hub get deployments -o name
INFO:root:Executing kubectl rollout status --namespace edsc-hub --watch deployment.extensions/hub
deployment "hub" successfully rolled out
INFO:root:Executing kubectl rollout status --namespace edsc-hub --watch deployment.extensions/proxy
deployment "proxy" successfully rolled out
INFO:root:Executing kubectl rollout status --namespace edsc-hub --watch deployment.extensions/user-scheduler
deployment "user-scheduler" successfully rolled out
```

### CODE REVIEW

File: user-images/edsc-hub/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM earthlab/earth-analytics-python-env:84549e0
+FROM earthlab/earth-analytics-python-env:f880467
```
Comment (lwasser) [inline user-images/edsc-hub/Dockerfile:1]: Here i update the ea python envt tag and i remove nbgrader stuff

File: hub-charts/edsc-hub/values.yaml [hunk 2]
```diff
@@ -21,11 +21,11 @@ jupyterhub:
       tag: set-on-deployment
     startTimeout: 600
     cpu:
-      guarantee: 0.1
-      limit: 1.
+      guarantee: 0.4
```
Comment (lwasser) [inline hub-charts/edsc-hub/values.yaml:24]: these are the old updates i made a few weeks ago to the resources on the hub

**Comment by kcranston** [comment]
So, I think I understand what's going on with the local vs travis. Whether or not to build (and optionally, push) the image depends on whether the image directory has changed. This is determined using `git diff` on the `user-images/<hubname>` directory, over a range of commits. That range comes from a [travis environment variable](https://github.com/earthlab/hub-ops/blob/master/deploy.py#L289) which is always going to be None when running `deploy.py` locally. 

I strongly suspect that the deploy script was never meant to be run locally.

**Comment by lwasser** [comment]
closing as this was an OLD pr

**Pull request closed without merging**